File: Lab5/Lab5/Lab5/Func.cs
using System;
using System.Linq;

namespace Lab5
{
    public class Func
    {
        //перевірка прямих масиву на першу вказану властивість 
        public static bool PointCheck(Line[] lines, int[] point, ref int index)
        {
            for (int i = 1; i < lines.Length; i++)
                if (lines[i].IsPerpendicular(lines[0]))
                    if (lines[i].PointBelongs(point))
                    {
                        index = i;
                        return true;
                    }
            return false;
        }
        
        //перевірка прямих масиву на другу вказану властивість 
        public static bool LineCheck(Line[] lines, ref int index)
        {
            for (int i = 0; i < lines.Length; i++)
            {
                bool exists = false;
                foreach (var line in lines)
                {
                    if (lines[i] == line) continue;
                    exists = lines[i].IsPerpendicular(line);
                }
                if (exists)
                {
                    index = i;
                    return true;
                }
            }
            return false;
        }
        
        //виведення масиву прямих
        public static void LinesOut(Line[] lines, string comment)
        {
            Console.WriteLine(comment);
            for (int i = 0; i < lines.Length; i++)
                Console.WriteLine($"Line #{i + 1}:\t{lines[i]}");
            Console.WriteLine();
        }
        
        //перетворення координат точки у рядок
        public static string PointString(int[] point)
        {
            return $"({string.Join("; ", point.Select(x => x.ToString()).ToArray())})";
        }
    }
}

File: Lab5/Lab5_py/func.py

# перевірка прямих списку на першу вказану властивість
def point_check(lines, point):
    for i in range(len(lines)):
        if lines[i].is_perpendicular(lines[0]):
            if lines[i].point_belongs(point):
                index = i
                return True, index
    return False, 0


# перевірка прямих списку на другу вказану властивість
def line_check(lines):
    for i in range(len(lines)):
        exists = False
        for line in lines:
            if lines[i] == line:
                continue
            exists = lines[i].is_perpendicular(line)
        if exists:
            index = i
            return True, index
    return False, 0


# виведення масиву прямих
def lines_out(lines, comment):
    print(comment)
    for i in range(len(lines)):
        print(f"Line #{i + 1}:\t{lines[i]}")
    print()


# перетворення координат точки у рядок
def point_string(point):
    point_str = [str(i) for i in point]
    return f"({'; '.join(point_str)})"
